openlibrary/plugins/worksearch/code.py

```diff
@@ -1018,23 +1018,28 @@ def work_search(query, sort=None, page=1, offset=0, limit=100, fields='*', facet
     query['wt'] = 'json'
     if sort:
         sort = process_sort(sort)
+
+    if 'availability' in fields and 'ia' not in fields:
+        fields.append('ia')
```

Comment:
Fields is a CSV; this causes an error
```suggestion
    if 'availability' in fields and 'ia' not in fields:
        fields += ",ia"
```

Reply:
Nope, nevermind; looks like it's sometimes a string -_-

Reply:
Not sure; this isn't working correctly on testing as is.